x/chainlink/keeper/msg_server.go

```diff
@@ -23,6 +23,16 @@ func (k Keeper) SubmitFeedDataTx(c context.Context, msg *types.MsgFeedData) (*ty
 		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, ErrIncorrectHeightFound)
 	}
 
+	feed := k.GetFeed(ctx, msg.FeedId)
+	reward := feed.GetFeed().FeedReward
+
+	rewardTokenAmount := types.NewLinkCoinInt64(int64(reward))
```

Comment:
I assume the `token bridge` will allow data providers to withdraw their reward?

Reply:
we might need to figure out what sort of bridge we will use though. right now, we are assuming the data provider will set up a cosmos account + node to submit data. will need to discuss and figure out more complex transfers